apps/antalmanac/src/components/Map/Map.tsx

```diff
@@ -41,7 +41,9 @@ interface MarkerContent {
 export function getCoursesPerBuilding() {
     const courseEvents = AppStore.getCourseEventsInCalendar();
 
-    const allBuildingCodes = courseEvents.map((event) => event.bldg.split(' ').slice(0, -1).join(' '));
+    const allBuildingCodes = courseEvents
+        .map((event) => event.bldg.map((location) => location.building))
+        .reduce((prev, cur) => prev.concat(cur), []);
```

Comment:
Can we just replace this reduce with a `flat()` call? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat

Reply:
Oh, I didn't realize what it was doing; I was just copying the code lol. But I can do you one better with `flatMap`

```suggestion
        courseEvents.flatMap(event => event.bldg.map(location => location.building))
```